src/java/fr/paris/lutece/plugins/workflow/modules/appointmentants/service/TaskAntsAppointmentService.java

```diff
@@ -0,0 +1,301 @@
+
+		try {
+			response = TaskAntsAppointmentRest.getAntsAppointmentStatus( getStatusUrl, token );
+		}
+		catch ( HttpAccessException h )
+		{
+			AppLogService.error( BEAN_SERVICE, h );
+		}
+		catch( Exception e )
+		{
+			AppLogService.error( BEAN_SERVICE, e );
+		}
+
+		if( response != null && !response.isEmpty( ) )
+		{
+			try
+			{
+				AntsStatusResponsePOJO responseObject = getStatusResponseAsObject( response );
+
+				/* If the application number hasn't been validated, or if it already has
+				 * appointments tied to it, then we shouldn't create any appointment
+				 * */
+				if ( !responseObject.getStatus().equals( validated ) ||
+						responseObject.getAppointments().length > 0)
+				{
+					return false;
+				}
+			}
+			catch( IOException e)
+			{
+				AppLogService.error( BEAN_SERVICE, e );
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Build the URL used to get the status of specific ANTS appointments
+	 * 
+	 */
+	public static String buildAntsGetStatusAppointmentUrl( List<String> applicationIdsList )
+	{
+		StringBuilder antsApisUrl = new StringBuilder(
+				AppPropertiesService.getProperty( TaskAntsAppointmentRestConstants.ANTS_URL ) ).
+				append( AppPropertiesService.getProperty( TaskAntsAppointmentRestConstants.ANTS_URL_STATUS_APPOINTMENT ) ).
+				append( "?" );
+
+		for( String applicationId : applicationIdsList )
+		{
+			antsApisUrl.append( AppPropertiesService.getProperty( TaskAntsAppointmentRestConstants.ANTS_APPLICATION_IDS ) ).
+			append( "=" ).append( applicationId ).
+			append( "&" );
+		}
+		// Remove the trailing character: '&'
+		return antsApisUrl.toString( ).substring( 0, antsApisUrl.length( ) - 1 );
+	}
+
+	/**
+	 * Creates an {@link AntsStatusResponsePOJO} Object from a json String containing the status
+	 * and appointments list that were returned by the ANTS API
+	 * 
+	 */
+	public static AntsStatusResponsePOJO getStatusResponseAsObject( String response ) throws IOException
+	{
+		ObjectMapper mapper = new ObjectMapper();
+		JsonNode jsonNode = mapper.readTree( response );
+
+		Iterator<String> fieldNames = jsonNode.fieldNames();
+
+		String fieldName = fieldNames.next( );
+
+		JsonNode field = jsonNode.get( fieldName );
+
+		return mapper.readerFor( AntsStatusResponsePOJO.class )
+				.readValue(field.toString());
+	}
+
+	/**
+	 * Properly encode a String containing a URL to make sure it has the proper format
+	 */
+	public static String cleanUrl( String urlToClean )
+	{
+		try
+		{
+			return URLEncoder.encode( urlToClean, "utf-8" );
+		}
+		catch( UnsupportedEncodingException e)
+		{
+			AppLogService.error( BEAN_SERVICE, e );
+			return urlToClean;
+		}		
+	}
+
+	@Transactional( WorkflowAppointmentAntsPlugin.BEAN_TRANSACTION_MANAGER )
+	@Override
+	public String getAntsApplicationFieldName( int idTask )
+	{
+		return _task_add_appointment_dao.load( idTask ).getAntsApplicationNumberFieldName( );
+	}
+
+	@Transactional( WorkflowAppointmentAntsPlugin.BEAN_TRANSACTION_MANAGER )
```

Comment:
Please use the transaction manager provided by lutece-core (TransactionManager.java) and remove the Spring annotation.